community/cypher/cypher-planner-3.4/src/main/scala/org/neo4j/cypher/internal/compiler/v3_4/planner/logical/steps/solveOptionalMatches.scala

```diff
@@ -54,7 +54,30 @@ case object outerHashJoin extends OptionalSolver {
         joinNodes == hint.variables.map(_.name).toSet
       }
 
-      Some(context.logicalPlanProducer.planOuterHashJoin(joinNodes, lhs, rhs, solvedHints, context))
+      Some(context.logicalPlanProducer.planLeftOuterHashJoin(joinNodes, lhs, rhs, solvedHints, context))
+    } else {
+      None
+    }
+  }
+}
+
+case object rightOuterHashJoin extends OptionalSolver {
+  override def apply(optionalQg: QueryGraph, rhs: LogicalPlan, context: LogicalPlanningContext, solveds: Solveds, cardinalities: Cardinalities) = {
```

Comment:
The code here looks very similar to the code above in the same file. Are there not opportunities for sharing it?